### ISSUE TITLE
Add image scaling\Current source compiles but crashes (solved)

### ISSUE THREAD

**Issue by ShoesForClues** [rxi/lovedos#10]
It would be great if this had the ability to draw bitmap at different sizes. I'd be fine with just having nearest neighbor.

**Comment by ShoesForClues** [comment]
I actually modified the code to allow image scaling. Though, while it compiles, it crashes in DOS. This was actually happening before I've made any changes to the source.

![](https://i.imgur.com/ghQvhGg.png)

**Comment by hseiken** [comment]
Are you using DOSBox or a real machine?  If you're using a real machine, what are the specs?  If you're using DOSBox, what specs are you using? 

To be honest, I won't be able to answer any questions about your issue, but I'm curious as to whether real machines are being used or simulations; I've run across a lot of bizarre crash anomalies with DOSBox on non-Love stuff as it is, so I don't really consider it a good emulator in general. I only ask about the real hardware specs as the videocard can be an issue and I'm not privvy to know if the LOVE graphics engine is all software-processor based or it if uses any special SVGA calls or anything of that nature as I remember different cards even back then had quirks of all kinds, too....

**Comment by ShoesForClues** [comment]
Yeah I'm using DOSBox. But I doubt it's an issue with the emulator because the precompiled version from the release seems to work fine. And lovedos does use software rendering. I don't actually know any DOS games that uses hardware acceleration.

And speaking of SVGA, I recently added VESA graphics support, and completely ditched the VGA mode. Mostly so I can just utilize higher resolutions than 320x200. And I can already tell that it isn't the graphics that's causing the unusual crash since I've compiled my own small test program for SVGA.


![](https://i.imgur.com/d0vOwvL.png)

**Comment by ShoesForClues** [comment]
I found the cause to why the program is crashing. It's because of the unsigned int's in his source code.

**Comment by hseiken** [comment]
VESA/SVGA are indeed accelerators, but not in the sense that we think of them today, obviously.  

Glad you found the problem with the unsigned numbers.  I only posited that it might have been DOSBox itself because I've found quite a lot of software that is non-gaming that doesn't work correctly under DOSBox due to so-called 'obscure practices'. Usually it's undocumented functions of the chipsets or timing not accurately being implemented within the emulator. For instance, Smalltalk/V for DOS works quite well on my real system, but in DOSBox it's graphics are really buggy, to the point where trying to fix it within the VM is nearly impossible.  

Anyway, cool that it's sorted out. I think LOVE for DOS is a super awesome idea and am looking forward to it improving and adding scaling is definitely an improvement.  :D  Good luck and sorry for being long-winded in my responses.

**Comment by ShoesForClues** [comment]
I've tried compiling version 0.2.0, but it seems like that one also crashes when I try to run it. Only reason why I assume that an unsigned integer is causing this crash is because I was able to replicate the same error when working on my own code from scratch. Though, I don't actually know which part of the source is causing it as there's numerous unsigned integers everywhere.

I'd say there could be a chance that it's a fault in the compiler and not the code itself, except I've tried 4 other different versions of djgpp, all resulting the same.

**Comment by hseiken** [comment]
I wish I could look into the problem, but truth be told, I'm more on the theoretical understanding of things and a bit of a 'fan' (much like Dan Carlin isn't a historian, but a self proclaimed 'fan of history').  I do some coding here and there, but low level stuff is beyond me when it comes to actually getting things working.

Other than offer the 'duh, I've tried that' armchair quarterback solutions like double checking spelling and capitalizations, etc. is to possibly look at the functions calling the faulty symbol. They might be unnecessarily running procedures that don't apply causing the compiling error...?  Also, linked libraries I've found always end up having weird issues with how they're supposed to be used and even different versions might change that.  Again, I've not looked at the source at all and I'm not that smart.

So...if nothing else, me sounding dumb out here will get a real nuts-and-bolts programmer out of the woodwork to help you simply to keep this noise from happening over here.  :)
